### PR TITLE
KYLIN-4688 Too many tmp files in HDFS tmp dictionary

### PR DESCRIPTION
**Pull request by ITzhangqiang**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: storage-hbase/src/main/java/org/apache/kylin/storage/hbase/steps/CubeHFileJob.java [hunk 1]
```diff
@@ -101,6 +102,7 @@ public int run(String[] args) throws Exception {
 
             // Automatic config !
             HFileOutputFormat3.configureIncrementalLoad(job, htable);
+            tmpPartitionFilePath = new Path(TotalOrderPartitioner.getPartitionFile(job.getConfiguration()));
```
Comment (zzcclp) [inline storage-hbase/src/main/java/org/apache/kylin/storage/hbase/steps/CubeHFileJob.java:105]: The value get from 'TotalOrderPartitioner.getPartitionFile' will be changed in method 'reconfigurePartitions', are you sure that the value of  'tmpPartitionFilePath' is right?

File: core-common/src/main/java/org/apache/kylin/common/util/HadoopUtil.java [hunk 2]
```diff
@@ -69,13 +68,6 @@ public static Configuration getCurrentConfiguration() {
     }
 
     public static Configuration healSickConfig(Configuration conf) {
-        // https://issues.apache.org/jira/browse/KYLIN-953
```
Comment (zzcclp) [inline core-common/src/main/java/org/apache/kylin/common/util/HadoopUtil.java:72]: There are some places calling this method, please check whether impact others.

File: storage-hbase/src/main/java/org/apache/kylin/storage/hbase/steps/CubeHFileJob.java (hunk 1 above)
Comment (zzcclp) [reply storage-hbase/src/main/java/org/apache/kylin/storage/hbase/steps/CubeHFileJob.java:105]: Got it.

**Comment by zzcclp** [comment]
I hava tested this pr on docker, it worked fine, test steps:
1. when building a segment, check the path '/tmp/hbase-staging', there were some 'partitions_' files;
2. when build successfully, the '/tmp/hbase-staging/partitions_' files were deleted; 
3. set property 'hbase.fs.tmp.dir' to '/tmp/hbase-tmp', and re-run step 1 and step 2,  the 'partitions_' which were in '/tmp/hbase-tmp' were deleted when build successfully, it worked fine.

LGTM, and @hit-lacus please review

**Comment by hit-lacus** [comment]
LGTM

**Pull request merged**